ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -467,36 +467,53 @@ module Intel = struct
       List.nth device.Pci.Pci_dev.size 2
       |> Int64.of_nativeint
     in
-    List.map
+    let rec filter_none acc lst =
+      match lst with
```

Comment:
This can be written as 
```
let rec filter_none acc = function 
  | [] -> ...
  | ...
```

You obviously did not want to use `List.filter ... |> List.map ...` to define it which is a little less efficient because of the intermediate lists that it would generate.

```
let filter_none xs = List.filter ((<>) None) xs |> List.map Opt.unbox